zipline/utils/cache.py

```diff
@@ -104,11 +100,210 @@ def __init__(self, cache=None):
             self._cache = {}
 
     def get(self, key, dt):
+        """Get the value of a cached object.
+
+        Parameters
+        ----------
+        key : any
+            The key to lookup.
+        dt : datetime
+            The time of the lookup.
+
+        Returns
+        -------
+        result : any
+            The value for ``key``.
+
+        Raises
+        ------
+        KeyError
+            Raised if the key is not in the cache or the value for the key
+            has expired.
+        """
         try:
             return self._cache[key].unwrap(dt)
         except Expired:
             del self._cache[key]
             raise KeyError(key)
 
     def set(self, key, value, expiration_dt):
+        """Adds a new key value pair to the cache.
+
+        Parameters
+        ----------
+        key : any
+            The key to use for the pair.
+        value : any
+            The value to store under the name ``key``.
+        expiration_dt : datetime
+            When should this mapping expire? The cache is considered invalid
+            for dates **strictly greater** than ``expiration_dt``.
+        """
         self._cache[key] = CachedObject(value, expiration_dt)
+
+
+class dataframe_cache(MutableMapping):
+    """A disk-backed cache for dataframes.
+
+    This object may be used as a context manager to delete the cache directory
+    on exit.
+
+    Parameters
+    ----------
+    path : str, optional
+        The directory path to the cache. Files will be written as
+        ``path/<keyname>``.
+    lock : Lock, optional
+        Thread lock for multithreaded/multiprocessed access to the cache.
+        If not provided no locking will be used.
+    clean_on_failure : bool, optional
+        Should the directory be cleaned up if an exception is raised in the
+        context manager.
+    serialize : {'msgpack', 'pickle'}, optional
+        How should the data be serialized.
```

Comment:
numpy does not suggest including the defaults because you can see that if you inspect the function signature. The only time that the default should be included is if it is lazy, like `None` which creates an empty dict or something.